### ISSUE TITLE
illegal hardware instruction when running tests on a fresh install on OS X

### ISSUE THREAD

**Issue by erip** [pytorch/torcharrow#161]
Following the instructions in the README to build and test torcharrow leads to an illegal hardware instruction error on OS X 10.15.7. I'm running on 2.9 GHz Dual-Core Intel Core i7 (i386). 

```sh
(torcharrow-dev) ➜  torcharrow git:(main) python -m unittest -v

test_ArrayColumnInt64 (torcharrow.test.lib_test.test_column.TestArrayColumns) ... [1]    29649 illegal hardware instruction  python -m unittest -v
```

**Comment by erip** [comment]
It seems like some of the guts of the code have an implicit dependency on a `MACOSX_DEPLOYMENT_TARGET=11`. I might be out of luck on this machine. 😅

**Comment by wenleix** [comment]
@erip : This is due to previously Velox/TrochArrow is built with `arch=native` that causes GCC very aggressively use AVX512. We have fixed it in https://github.com/facebookresearch/torcharrow/pull/315 (in fact the recent portability improvement in Velox/TorchArrow should allow it to work with SSE-only machines). Would you mind try again (or just try the nightly build). Thanks!

**Comment by wenleix** [comment]
Believe this is fixed via #315 . Feel free to reopen it if it still encounters .
